### DISCUSSION TITLE
What is the best way to display a TypeScript type ?

### DISCUSSION THREAD

**Discussion by JesusTheHun** [vuejs/vitepress#3744] [General]
This is general question, not really specific to Vitepress (but tied to its features and limitations).
How do you present, document, a TypeScript type ?

In my documentation I want to introduce the user to some types that do some cool stuff and can help him when using my library, but I always struggle to define what is the best way. Half the time I can come up with an example of use case, but it does not always suffice to demonstrate the benefits.

So far I've used the _comment queries_ style, but I'm wondering if there is a better way.

```ts
const foo = await bar('yolo');
//     ?^ SomeKindOfResult<'yolo'>
```

**Comment by brc-dd** [comment]
shiki twoslash - https://shiki.style/packages/vitepress

**Comment by JesusTheHun** [comment]
@brc-dd that's pretty solid !
I'm struggling to build when referencing a third party package though. Well, it's actually a package in the same monorepo. I've tried to work with `fsMap` and such, but if you have a working recipe, it would save me a lot of time :)

EDIT : nvm, it was a build mistake on netlify
